## 7. Parameter Estimation

**Exercise 7.1**. From a series of length 100, we have computed $r_1 = 0.8$, $r_2 = 0.5$, $r_3 = 0.4$, $\overline{Y} = 2$, and a sample variance of 5. If we assume that an AR(2) model with a constant term is appropriate, how can we get (simple) estimates of $\phi_1$, $\phi_2$, $\theta_0$, and $\sigma_e^2$?

**Solution**.  Equation (7.1.2) gives us, for the AR(2) model, the estimates

$$ \hat{\phi}_1 = \frac{r_1(1 - r_2)}{1 - r_1^2} \quad \text{and} \quad \hat{\phi}_2 = \frac{r_2 - r_1^2}{1 - r_1^2} $$

Then,

$$ \hat{\theta}_0 = \overline{Y}(1 - \hat{\phi}_1 - \hat{\phi}_2) $$

and from Equation (7.1.8)

$$ \hat{\sigma}_e^2 = s^2(1 - \hat{\phi}_1 r_1 - \hat{\phi}_2 r_2) $$

In [1]:
import numpy as np

In [2]:
r1, r2, r3 = 0.8, 0.5, 0.4
Ybar = 2
s2 = 5

In [3]:
phi1 = r1 * (1 - r2) / (1 - r1**2)
phi2 = (r2 - r1**2) / (1 - r1**2)
theta0 = Ybar * (1 - phi1 - phi2)
se = s2 * (1 - phi1 * r1 - phi2 * r2)

In [4]:
print((phi1, phi2, theta0, se))

(1.1111111111111116, -0.3888888888888894, 0.5555555555555556, 1.527777777777777)


**Exercise 7.2**.  Assuming that the following data arise from a stationary process, calculate method-of-moments estimates of $\mu$, $\gamma_0$, and $\rho_1$: 6, 5, 4, 6, 4.

**Solution**.

In [5]:
Y = np.array([6, 5, 4, 6, 4])

In [6]:
mu_hat = np.mean(Y)
gamma_0_hat = np.sum((Y - mu_hat)**2) / (len(Y) - 1)
rho_1 = (Y[:-1] - mu_hat)@(Y[1:] - mu_hat) / (len(Y) - 1)

print((mu_hat, gamma_0_hat, rho_1))

(5.0, 1.0, -0.5)


**Exercise 7.3**. If $\{Y_t\}$ satisfies an AR(1) model with $\phi$ of about 0.7, how long of a series do we need to estimate $\phi = \rho_1$ with 95% confidence that our estimation error is no more than $\pm 0.1$?

**Solution**.  For the AR(1) model, the large sample standard error of $\hat{\phi}$ is $\sqrt{(1 - \phi^2) / n}$.  For a degree of confidence of $1 - \alpha$, we need

$$ \Phi(1 - \alpha/2) \sqrt{\frac{1 - \phi^2}{n}} \leq 0.1 $$

or

$$ (1 - \phi^2) \left( \frac{\Phi(1 - \alpha/2)}{0.1} \right)^2 \leq n $$

where $\Phi$ is the CDF of the standard normal.  Replacing in $\alpha = 0.05$ and $\phi = 0.7$, the bound becomes $n \geq 196$.

**Exercise 7.4** Consider an MA(1) process for which it is *known* that the process mean is zero.  Based on a series of length $n = 3$, we observe $Y_1 = 0$, $Y_2 = −1$, and $Y_3 = 1/2$.

**(a)** Show that the conditional least-squares estimate of $\theta$ is $1/2$.

**(b)** Find an estimate of the noise variance. (Hint: Iterative methods are not needed in this simple case.)

**Solution**.

**(a)**  From Equation (7.2.14),

$$
\begin{array}{rcl}
e_1 &=& Y_1 \\
e_2 &=& Y_2 + \theta e_1 \\
e_3 &=& Y_3 + \theta e_2 \\
\end{array}
$$

and so $e_1 = Y_1 = 0$, $e_2 = Y_2 + \theta e_1 = -1$, and $e_3 = Y_3 + \theta e_2 = 1/2 - \theta$.  The conditional sum-of-squares is

$$ S_c(\theta) = \sum_i e_i^2 = 0^2 + 1^2 + \left( \frac{1}{2} - \theta \right)^2 $$

which is minimized at $\theta = 1/2$.

**(b)**  From Equation (7.1.9),

$$ \hat{\sigma}_e^2(\theta) = \frac{s^2}{1 + \theta^2} = \frac{1}{1 + \theta^2} \left( \frac{1}{n - 1} S_c(\theta) \right) $$

which, for $\theta = 1/2$, has value $\sigma_e^2 = 0.4$.

**Exercise 7.5**.  Given the data $Y_1 = 10$, $Y_2 = 9$, and $Y_3 = 9.5$, we wish to fit an IMA(1,1) model without a constant term.

**(a)** Find the conditional least squares estimate of $\theta$. (Hint: Do Exercise 7.4 first.)

**(b)** Estimate $\sigma_e^2$.

**Solution**.

**(a)**  We have $\nabla Y_1 = -1$ and $\nabla Y_2 = 0.5$.  Fitting the MA(1) model on $\nabla Y_t$ with a zero mean,

$$
\begin{array}{rcl}
e_1 &=& \nabla Y_1 \\
e_2 &=& \nabla Y_2 + \theta e_1
\end{array}
$$

and so $e_1 = \nabla Y_1 = -1$ and $e_2 = \nabla Y_2 + \theta e_1 = 0.5 - \theta$.  The conditional sum-of-squares is

$$ S_c(\theta) = \sum_i e_i^2 = (-1)^2 + (0.5 - \theta)^2 $$

which is minimized at $\theta = 0.5$.

**(b)**  From Equation (7.1.9),

$$ \hat{\sigma}_e^2(\theta) = \frac{s^2}{1 + \theta^2} = \frac{1}{1 + \theta^2} \left( \frac{1}{n - 1} S_c(\theta) \right) $$

which, for $\theta = 0.5$, has value $\hat{\sigma}_e^2 = 0.8$.

**Exercise 7.6**.  Consider two different parameterizations of the AR(1) process with nonzero mean:

$$
\begin{array}{lrcl}
\text{Model I:}  &  Y_t − \mu &=& \phi(Y_{t−1} − \mu) + e_t \\
\text{Model II:} &         Y_t &=& \phi Y_{t−1} + \theta_0 + e_t \\
\end{array}
$$

We want to estimate $\phi$ and $\mu$ or $\phi$ and $\theta_0$ using conditional least squares conditional on $Y_1$. Show that with Model I we are led to solve nonlinear equations to obtain the estimates, while with Model II we need only solve linear equations.

**Solution**.  We can express Model I as

$$ Y_t = \phi Y_{t-1} + \mu (1 - \phi) + e_t $$

which is non-linear on $\mu$ and $\phi$, and so setting the partial derivatives of $S_c(\mu, \phi) = \sum_t (e_t)^2$ to zero will not be linear equations.

On the other hand, model II is linear on $\phi$ and $\theta_0$, so setting the partial derivatives of $S_c(\mu, \phi) = \sum_t (e_t)^2$ to zero produces linear equations on $\phi$ and $\theta_0$.

**Exercise 7.7**. Verify Equation (7.1.4) on page 150.

**Solution**.  Equation (7.1.4) states that, for the MA(1) process, the root satisfying the invertibility condition $|\theta| < 1$ has estimate

$$ \hat{\theta} = \frac{-1 + \sqrt{1 - 4r_1^2}}{2r_1} $$

From Equation (4.2.2), we have

$$ r_1 = - \frac{\hat{\theta}}{1 + \hat{\theta}^2} $$

which can be written as the second-degree equation

$$ \hat{\theta}^2 r_1 + \hat{\theta} + r_1 = 0 $$

Its roots are

$$ \frac{-1 \pm \sqrt{1 - 4 r_1^2}}{2r_1} $$

The product of the roots is 1, so only one of them can have absolute value less than or equal to 1.  We also have

$$ \left|\frac{-1 + \sqrt{1 - 4r_1^2}}{2r_1} \right| < \frac{1}{| 2 r_1 |} \leq 1 $$

and so the given root should be the selected estimate.

**Exercise 7.8**.  Consider an ARMA(1,1) model with $\phi = 0.5$ and $\theta = 0.45$.

**(a)** For $n = 48$, evaluate the variances and correlation of the maximum likelihood estimators of $\phi$ and $\theta$ using Equations (7.4.13) on page 161. Comment on the results.

**(b)** Repeat part (a) but now with $n = 120$. Comment on the new results.

**Solution**.

**(a)**  Equations (7.4.13) state that:

$$
\begin{array}{rcl}
\text{Var}[\hat{\phi}] &\approx& \left[ \frac{1 - \phi^2}{n} \right] \left[ \frac{1 - \phi \theta}{\phi - \theta}\right]^2 \\
\text{Var}[\hat{\theta}] &\approx& \left[ \frac{1 - \theta^2}{n} \right] \left[ \frac{1 - \phi \theta}{\phi - \theta}\right]^2 \\
\text{Corr}[\hat{\phi}, \hat{\theta}] &\approx& \frac{\sqrt{(1 - \phi^2)(1 - \theta^2)}}{1 - \phi \theta}
\end{array}
$$

In [7]:
phi = 0.5
theta = 0.45
n = 48

var_phi_hat = ((1 - phi**2)/n) * ((1 - phi * theta) / (theta - phi))**2
var_theta_hat = ((1 - theta**2)/n) * ((1 - phi * theta) / (theta - phi))**2
corr_phi_hat_theta_hat = np.sqrt((1 - phi**2) * (1 - theta**2)) / (1 - phi * theta)

var_phi_hat, var_theta_hat, corr_phi_hat_theta_hat

(3.7539062500000018, 3.9916536458333347, 0.9979166644037433)

The variables are close to each other, which causes the variances to be high and the estimates to be highly correlated.

**(b)**

In [8]:
phi = 0.5
theta = 0.45
n = 120

var_phi_hat = ((1 - phi**2)/n) * ((1 - phi * theta) / (theta - phi))**2
var_theta_hat = ((1 - theta**2)/n) * ((1 - phi * theta) / (theta - phi))**2
corr_phi_hat_theta_hat = np.sqrt((1 - phi**2) * (1 - theta**2)) / (1 - phi * theta)

var_phi_hat, var_theta_hat, corr_phi_hat_theta_hat

(1.5015625000000008, 1.596661458333334, 0.9979166644037433)

The variances of the estimates go down with $n$, but the correlation does not.

**Exercise 7.9**. Simulate an MA(1) series with $\theta = 0.8$ and $n = 48$.

**(a)** Find the method-of-moments estimate of $\theta$.

**(b)** Find the conditional least squares estimate of $\theta$ and compare it with part (a).

**(c)** Find the maximum likelihood estimate of $\theta$ and compare it with parts (a) and (b).

**(d)** Repeat parts (a), (b), and (c) with a new simulated series using the same parameters and same sample size. Compare your results with your results from the first simulation.

**Solution**.

In [9]:
import numpy as np
from statsmodels.tsa.arima_process import ArmaProcess
from statsmodels.tsa.arima_model import ARIMA

def generate_arima(phi=[], d=0, theta=[], n=100):
    ar = np.r_[1, -np.array(phi)]
    ma = np.r_[1, -np.array(theta)]
    Y = ArmaProcess(ar, ma).generate_sample(nsample=n)
    for i in range(d):
        Y = np.cumsum(Y)
    return Y

In [10]:
np.random.seed(0)
Y = generate_arima(theta=[0.8], n=48)

**(a)**

In [11]:
from statsmodels.tsa.stattools import acf

def estimate_ma1_mom(x):
    r = acf(Y, fft=False, nlags=1)[1]
    if np.abs(r) < 0.5:
        return (-1 + np.sqrt(1 - 4 * r**2))/(2*r)
    return np.nan

In [12]:
estimate_ma1_mom(Y)

0.6403760599885285

**(b)**

In [13]:
ARIMA(Y, order=(0, 0, 1)).fit(method='css').maparams

array([-0.75869964])

With our convention for the sign of $\theta$, this estimate is $\theta=0.759$, which is a better estimate than in part (a).

**(c)**

In [14]:
ARIMA(Y, order=(0, 0, 1)).fit(method='mle').maparams

array([-0.99986536])

With our convention for the sign of $\theta$, this estimate is very close to $\theta=1$, which is concerning as it does not provide invertibility.

**(d)**

In [15]:
np.random.seed(1)
Y = generate_arima(theta=[0.8], n=48)
print(estimate_ma1_mom(Y))
print(ARIMA(Y, order=(0, 0, 1)).fit(method='css').maparams)
print(ARIMA(Y, order=(0, 0, 1)).fit(method='mle').maparams)

nan
[-0.99999996]
[-0.99995614]


This time, the method of moments failed to provide a valid estimate, while both the CSS and MLE methods provided an estimate very close to 1; results are worse.

**Exercise 7.10**.  Simulate an MA(1) series with $\theta = −0.6$ and $n = 36$.

**(a)** Find the method-of-moments estimate of $\theta$.

**(b)** Find the conditional least squares estimate of $\theta$ and compare it with part (a).

**(c)** Find the maximum likelihood estimate of $\theta$ and compare it with parts (a) and (b).

**(d)** Repeat parts (a), (b), and (c) with a new simulated series using the same parameters and same sample size. Compare your results with your results from the first simulation.

**Solution**.

In [16]:
np.random.seed(1000)
Y = generate_arima(theta=[-0.6], n=36)

**(a)**

In [17]:
estimate_ma1_mom(Y)

-0.36076887498825894

**(b)**

In [18]:
ARIMA(Y, order=(0, 0, 1)).fit(method='css').maparams

array([0.4716205])

With our convention for the sign of $\theta$, this estimate is very close to $\theta=1$, which is concerning as it does not provide invertibility.

**(c)**

In [19]:
ARIMA(Y, order=(0, 0, 1)).fit(method='mle').maparams

array([0.48058205])

With our convention for the sign of $\theta$, this estimate is very close to $\theta=1$, which is concerning as it does not provide invertibility.

**(d)**

In [20]:
np.random.seed(1001)
Y = generate_arima(theta=[-0.6], n=36)
print(estimate_ma1_mom(Y))
print(ARIMA(Y, order=(0, 0, 1)).fit(method='css').maparams)
print(ARIMA(Y, order=(0, 0, 1)).fit(method='mle').maparams)

nan
[0.66970618]
[0.64767655]


This time, method-of-moments failed to provide an estimate, while both CSS and ML provided estimates closer to the real value of $\theta = -0.6$.